packages/core-transaction-pool/lib/guard.js

```diff
@@ -133,7 +133,7 @@ module.exports = class TransactionGuard {
     transactions.forEach(transaction => {
       const exists = this.pool.transactionExists(transaction.id)
 
-      if (!exists && !this.pool.isSenderBlocked(transaction.senderPublicKey)) {
+      if (!exists && !this.pool.isSenderBlocked(transaction.senderPublicKey, transaction.id)) {
```

Comment:
in this case pass the whole transaction object to the function, so something like isSenderBlocked(transaction)

Reply:
this things are our of log for the reason of not filling the logs too much, but I see the usecase, and as @dated suggested - we can then move this up higher